src/formatDate.js

```diff
@@ -50,7 +51,44 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldDate = date.split(fromFormat[3]);
+  let newDate = [''];
+
+  if (fromFormat.indexOf('YYYY') !== -1) { // if we shorten YYYY to YY
+    if (toFormat.indexOf('YY') !== -1) {
+      toFormat[toFormat.indexOf('YY')] = 'YYYY';
+      oldDate[fromFormat.indexOf('YYYY')] = oldDate[fromFormat.indexOf('YYYY')].toString().slice(2);
+    }
+  }
+
+  if (fromFormat.indexOf('YY') !== -1) { // if we extend YY to YYYY
+    if (toFormat.indexOf('YYYY') !== -1) {
+      toFormat[toFormat.indexOf('YYYY')] = 'YY';
+
+      switch (true) {
+        case oldDate[fromFormat.indexOf('YY')] < '30': // if we extend YY < 30 to YYYY
+        case oldDate[fromFormat.indexOf('YY')] === '00': // if we extend YY = 00 to YYYY
+          oldDate[fromFormat.indexOf('YY')] = 20 + '' + oldDate[fromFormat.indexOf('YY')];
+          break;
+
+        case oldDate[fromFormat.indexOf('YY')] === '30': // if we extend YY = 30 to YYYY
+        case oldDate[fromFormat.indexOf('YY')] > '30': // if we extend YY > 30 to YYYY
+          oldDate[fromFormat.indexOf('YY')] = 19 + '' + oldDate[fromFormat.indexOf('YY')];
+          break;
+      }
+    }
+  }
```

Comment:
I think you can do this with two cycles and two swtich statements, if you want you can improve your solution